### PR TITLE
Add a new time_pattern option.

### PR DESCRIPTION
**Pull request by cpinkham**
- Default new option value to 'HH:mm'
- Remove time_format fallbacks to locale and instead fall through to using time_pattern.
- time_format option can be removed when ready leaving time_pattern to act for time the same as date_pattern acts for date.

References #240

### CODE REVIEW

File: src/clock-weather-card.ts [hunk 1]
```diff
@@ -440,19 +441,13 @@ export class ClockWeatherCard extends LitElement {
   }
 
   private time (date: DateTime = this.currentDate): string {
+    // Honor deprecated time_format setting until removed
```
Comment (pkissling) [inline src/clock-weather-card.ts:444]: in order not to introduce a breaking change, the precedence should be `time_pattern` -> `time_format` -> `default`, so please do not remove existing code, but add 

```
if (this.config.time_pattern) {
   return this.toZonedDate(date).toFormat(this.config.time_pattern)
}
```

as the first condition in function `time(...)`

File: src/types.ts [hunk 2]
```diff
@@ -39,6 +40,7 @@ export interface MergedClockWeatherCardConfig extends LovelaceCardConfig {
   forecast_rows: number
   locale?: string
   time_format?: '12' | '24'
+  time_pattern: string
```
Comment (pkissling) [inline src/types.ts:43]: should be optional

File: src/clock-weather-card.ts [hunk 3]
```diff
@@ -390,6 +390,7 @@ export class ClockWeatherCard extends LitElement {
       hourly_forecast: config.hourly_forecast ?? false,
       animated_icon: config.animated_icon ?? true,
       time_format: config.time_format?.toString() as '12' | '24' | undefined,
+      time_pattern: config.time_pattern ?? 'HH:mm',
```
Comment (pkissling) [inline src/clock-weather-card.ts:393]: fallback to `undefined` if not provided

File: README.md [hunk 4]
```diff
@@ -127,7 +127,8 @@ time_zone: null
...
-| time_format           | `24` \| `12`     | **Optional** | Format used to display the time. If not provided, falls back to the default time format of the configured `locale`                                                                                                                | `24`      |
+| time_format           | `24` \| `12`     | **Optional** | Format used to display the time. If not provided, falls through to time_pattern format                                                                                                                                            | `null`    |
+| time_pattern          | string           | **Optional** | Pattern to use for time formatting. If not provided, falls back to a localized default time formatting. See [luxon](https://moment.github.io/luxon/#/formatting?id=table-of-tokens) for valid tokens                              | `HH:MM`   |
```
Comment (pkissling) [inline README.md:131]: precedence `time_pattern` -> `time_format` -> localized default time formatting.
default value should be `null`.

please also add the newly introduced property in `### Full configuration`

**Review by pkissling — CHANGES_REQUESTED** [review]

**Comment by pkissling** [comment]
thanks for the PR and your contribution. i requested some changes to make sure we don't breaking existing installations of clock-weather-card after updating to the version including your changes

**Comment by cpinkham** [comment]
I have all of these ready but have one question about the Full Configuration change before I update the pull request.  Do you want it to be noted in any way in the Full Configuration section indicating time_pattern overrides time_format if both are provided?

Something like:

time_pattern: HH:mm
# Note: time_format is ignored if both time_pattern and time_format are provided
time_format: 24

Thanks for your feedback on this, this is my first time using TypeScript or contributing to HA, so it has been a slight learning curve.

**Comment by pkissling** [comment]
> I have all of these ready to add to a commit but have one question about the Full Configuration change before I update the pull request. Do you want it to be noted in any way in the Full Configuration section indicating time_pattern overrides time_format if both are provided?
> 
> Something like:
> 
> time_pattern: HH:mm # Note: time_format is ignored if both time_pattern and time_format are provided time_format: 24
> 
> Thanks for your feedback on this, this is my first time using TypeScript or contributing to HA, so it has been a slight learning curve.

no problem, i'm glad your are contributing new features.
i think its sufficient if you mention it in the corresponding property description of https://github.com/pkissling/clock-weather-card#options

**Comment by pkissling** [comment]
thanks for your contribution!

**Review by pkissling — APPROVED** [review]

**Comment by cpinkham** [comment]
Sorry, I hadn't commented again yet because I was going to see what I could do about that merge.  I wanted a cleaner request without all the extra commits listed.  Thanks for pulling this in.

**Pull request merged**